genesyscloud/resource_genesyscloud_webdeployments_configuration.go

```diff
@@ -0,0 +1,888 @@
+		} else {
+			if version > maxVersion {
+				maxVersion = version
+			}
+		}
+	}
+
+	if maxVersion == 0 {
+		return "draft"
+	}
+
+	return strconv.Itoa(maxVersion)
+}
+
+func readWebDeploymentConfiguration(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	api := platformclientv2.NewWebDeploymentsApiWithConfig(sdkConfig)
+
+	version := d.Get("version").(string)
+	log.Printf("Reading web deployment configuration %s", d.Id())
+	return withRetriesForRead(ctx, 30*time.Second, d, func() *resource.RetryError {
+		if version == "" {
+			version = determineLatestVersion(api, d.Id())
+		}
+		configuration, resp, getErr := api.GetWebdeploymentsConfigurationVersion(d.Id(), version)
+		if getErr != nil {
+			if isStatus404(resp) {
+				return resource.RetryableError(fmt.Errorf("Failed to read web deployment configuration %s: %s", d.Id(), getErr))
+			}
+			return resource.NonRetryableError(fmt.Errorf("Failed to read web deployment configuration %s: %s", d.Id(), getErr))
+		}
+
+		d.Set("name", *configuration.Name)
+		if configuration.Description != nil {
+			d.Set("description", *configuration.Description)
+		}
+		d.Set("languages", *configuration.Languages)
+		d.Set("default_language", *configuration.DefaultLanguage)
+		d.Set("status", *configuration.Status)
+		d.Set("version", *configuration.Version)
+		if configuration.Messenger != nil {
+			d.Set("messenger", flattenMessengerSettings(configuration.Messenger))
+		}
+		if configuration.JourneyEvents != nil {
+			d.Set("journey_events", flattenJourneyEvents(configuration.JourneyEvents))
+		}
+
+		log.Printf("Read web deployment configuration %s %s", d.Id(), *configuration.Name)
+		return nil
+	})
+}
+
+func updateWebDeploymentConfiguration(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name, inputCfg := readWebDeploymentConfigurationFromResourceData(d)
+
+	log.Printf("Updating web deployment configuration %s", name)
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	api := platformclientv2.NewWebDeploymentsApiWithConfig(sdkConfig)
+
+	_, _, err := api.PutWebdeploymentsConfigurationVersionsDraft(d.Id(), *inputCfg)
+	if err != nil {
+		return diag.Errorf("Error updating web deployment configuration %s: %s", name, err)
+	}
+
+	activeError := waitForConfigurationDraftToBeActive(ctx, api, d.Id())
+	if activeError != nil {
+		return diag.Errorf("Web deployment configuration %s did not become active and could not be published", name)
+	}
+
+	configuration, _, err := api.PostWebdeploymentsConfigurationVersionsDraftPublish(d.Id())
+	if err != nil {
+		return diag.Errorf("Error publishing web deployment configuration %s: %s", name, err)
+	}
+	d.Set("version", configuration.Version)
+	d.Set("status", configuration.Status)
+
+	log.Printf("Finished updating web deployment configuration %s", name)
+	//	time.Sleep(5 * time.Second)
```

Comment:
This could be uncommented. Eventual consistency usually causes issues when we update and try to read a resource straight away